estate/models/estate.py

```diff
@@ -0,0 +1,100 @@
+    salesperson_id = fields.Many2one("hr.employee", string="Salesperson", default=_get_default_salesperson_id)
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    best_price = fields.Float("Best Price", compute="_compute_best_price")
+    property_tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+    description = fields.Text("Description")
+    postcode = fields.Char("Postcode")
+    date_availability = fields.Date("Available From", copy=False, default=_get_default_date_availability)
+    expected_price = fields.Float("Expected Price", required=True, default=1e6)
+    selling_price = fields.Float("Selling Price", readonly=True, copy=False)
+    bedrooms = fields.Integer("Bedrooms", default=2)
+    living_area = fields.Integer("Living Area (sqm)")
+    total_area = fields.Integer("Total Area (sqm)", compute="_compute_total_area")
+    facades = fields.Integer("Facades")
+    garage = fields.Boolean("Garage")
+    garden = fields.Boolean("Garden")
+    garden_area = fields.Integer("Garden Area (sqm)")
+    garden_orientation = fields.Selection(string="Garden Orientation", selection=[
+        ("north", "North"),
+        ("south", "South"),
+        ("east", "East"),
+        ("west", "West"),
+    ])
+
+    state = fields.Selection(string="State", required=True, default="new", selection=[
+        ("new", "New"),
+        ("offer_received", "Offer Received"),
+        ("offer_accepted", "Offer Accepted"),
+        ("sold", "Sold"),
+        ("canceled", "Canceled"),
+    ])
+
+    active = fields.Boolean("Active", default=True)
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_price(self):
+        for record in self:
+            record.best_price = max(record.offer_ids.mapped("price"), default=0.)
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = self.garden_area or 10
+            self.garden_orientation = self.garden_orientation or "north"
+        else:
+            self.garden_area = 0
+            self.garden_orientation = None
+
+    @api.constrains("selling_price")
+    def _check_selling_price(self):
+        for record in self:
+            if float_is_zero(record.selling_price, precision_digits=2):
+                continue
+            if float_compare(record.expected_price * .9, record.selling_price, precision_digits=2) == 1:
+                raise ValidationError("Selling price cannot be lower than 90% of the expected price")
+
+    @util.action
+    def action_set_sold(self):
+        for record in self:
+            if record.state == "canceled":
+                raise UserError("A property can't be sold after it was canceled")
+            record.state = "sold"
```

Comment:
If this method is only reachable using a button on a form view, it is safe to assume that having a recordset with multiple elements would not make any sense.
```suggestion
    @util.action
    def action_set_sold(self):
            self.ensure_one()
            if self.state == "canceled":
                raise UserError("A property can't be sold after it was canceled")
            self.state = "sold"
```